src/Init/Data/List/BasicAux.lean

```diff
@@ -12,60 +12,138 @@ namespace List
 /-! The following functions can't be defined at `Init.Data.List.Basic`, because they depend on `Init.Util`,
    and `Init.Util` depends on `Init.Data.List.Basic`. -/
 
-def get! [Inhabited α] : List α → Nat → α
+/--
+Returns the `i`-th element in the list (zero-based).
+
+If the index is out of bounds (`i ≥ as.length`), this function panics when executed, and returns
+`default`. See `get?` and `getD` for safer alternatives.
```

Comment:
```suggestion
If the index is out of bounds (`i ≥ as.length`), this function panics when executed, crashing
in compiled programs and returning `default` from the perspective of proofs. See `List.get?`
and `List.getD` for safer alternatives.
```

Reply:
This is not true. It is important for users to realize that `panic!` does not halt execution, because it is often a vector for even worse issues when further operations are applied to the returned dummy value. In compiled programs, the effect of `panic!` is to print to the console, and then continue with the provided default value.

Reply:
Yeah, I think merely "panics" is the thing to write here. I certainly keep misleading myself about it - thanks for the correction, @digama0!